p2p/channel.go

```diff
@@ -0,0 +1,96 @@
+package p2p
+
+import (
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+// ChannelID is an arbitrary channel ID.
+type ChannelID uint16
+
+// Envelope specifies the message receiver and sender.
+type Envelope struct {
+	From      PeerID        // Message sender, or empty for outbound messages.
+	To        PeerID        // Message receiver, or empty for inbound messages.
+	Broadcast bool          // Send message to all connected peers, ignoring To.
+	Message   proto.Message // Payload.
+}
+
+// Channel is a bidirectional channel for Protobuf message exchange with peers.
```

Comment:
```suggestion
// Channel is a bidirectional channel for Protobuf message exchange with peers.
//
// Channel is safe for concurrent use by multiple goroutines.
```

do we want to specify that `Channel` is safe for concurrent use?